### DISCUSSION TITLE
How to schedule watchtower to run the first Sunday of every month?

### DISCUSSION THREAD

**Discussion by scaramouche1** [containrrr/watchtower#1387] [Q&A] [answered]
There is a hack-ish way to tell Cron to run a script the first Sunday of every month (see [here](https://stackoverflow.com/questions/3241086/how-to-schedule-to-run-first-sunday-of-every-month)).

Is there a way to tell Watchtower (installed using docker-compose) to run on that schedule?

**Comment by piksel** [comment, marked as answer]
This looks like it should work: `0 0 0 1-7 * sun`, but the spec explicitly handles restricting both DoM and DoW as "inclusive or":

> While normally the job is executed when the time/date specification fields all match the current time and date, there is one exception: if both "day of month" (field 3) and "day of week" (field 5) are restricted (not contain "*"), then one or both must match the current day.

Some implementations have added a new special character `#` for this, but that is unfortunately not supported by [the scheduling library](https://github.com/robfig/cron) we use. If it *was* supported, it would look like this: `0 0 0 ? * sun#1`.

So no, unfortunately, that does not seem possible right now.

**Comment by scaramouche1** [comment]
Thank you for the prompt response!  I see there are [several](https://github.com/robfig/cron/pull/325) [requests](https://github.com/robfig/cron/issues/434) to add this functionality to the underlying library, so let's hope it gets added soon.  At the same time, the project may have been [abandoned](https://github.com/robfig/cron/issues/463), so a move to a new library may be necessary.
